virtool/tasks/client.py

```diff
@@ -22,15 +27,26 @@ def __init__(self, redis: Redis):
         self.redis = redis
 
     async def enqueue(self, task_type: str, task_id: int):
-        logging.info(task_id)
         await self.redis.rpush(REDIS_TASKS_LIST_KEY, task_id)
 
     async def pop(self) -> int:
-        with await self.redis as redis:
-            result = await redis.blpop(REDIS_TASKS_LIST_KEY)
-
-            if result is not None:
-                return int(result[1])
+        result = await self._blpop(self.redis)
+
+        if result is not None:
+            return int(result[1])
+
+    @staticmethod
+    async def _blpop(redis: Redis):
```

Comment:
Why make this a static method? I'm not sure why it can't just reference to `self.redis`.

Please add brief docstring as to why this is necessary and what it's doing. It's not immediately clear to me.

Reply:
Ahh yeah, that's a holdover from when I was experimenting with a decorator based solution for wrapping redis commands we might want to retry. (something like `@retryRedisCommand`) I could never make the api work in a way I thought was intuitive and reliable so I dropped it.

Has been updated to get redis from self.